nameko/containers.py

```diff
@@ -348,11 +364,15 @@ def _run_worker(self, worker_ctx, handle_result):
 
         with _log_time('ran worker %s', worker_ctx):
 
-            self.dependencies.injections.all.inject(worker_ctx)
+            for dependency in self.dependencies:
```

Comment:
I've reverted to using an ``inject`` method on ``Dependency`` for now, which works but seems a bit backwards to me.

I find the ``SpawningSet`` a bit clunky really. I'd rather have a ``spawning_imap`` helper or similar that just accepts a callable. Would be more explicit.